modules/aerodyn/src/AeroDyn.f90

```diff
@@ -1681,29 +1675,88 @@ subroutine AD_CalcWind(t, u, p, o, m, ErrStat, ErrMsg)
    
    integer(intKi)                               :: ErrStat2
    character(ErrMsgLen)                         :: ErrMsg2
-   integer(intKi)                               :: node
+   integer(intKi)                               :: StartNode, iWT, k
+   real(ReKi)                                   :: PosOffset(3)
+   real(ReKi), allocatable                      :: NoAcc(:,:)
 
    ErrStat = ErrID_None
    ErrMsg = ""
-   
-   ! Set wind positions
-   node = 1 ! Index of first node in WindPos array
-   call AD_GetWindPositions(p, u, o, m%WindPos, node, errStat2, errMsg2)
-   if(Failed()) return
 
-   ! Calculate wind velocity/acceleration
-   call IfW_FlowField_GetVelAcc(p%FlowField, 1, t, m%WindPos, m%WindVel, m%WindAcc, errStat2, errMsg2)
-   if(Failed()) return
+   ! Initialize node. The StartNode is used for OpenFOAM to provide the wind
+   ! velocities. The node ordering in OpenFOAM must match that used in here.
+   StartNode = 1
 
-   ! Transfer wind velocity to nodes
-   node = 1  ! Index of first node in WindVel array
-   call AD_GetWindVel(u, m%WindVel, node, errStat2, errMsg2)
-   if(Failed()) return
+   do iWT = 1, size(u%rotors)
+
+      ! If rotor is MHK, add water depth to z coordinate
+      if (p%rotors(iWT)%MHK > 0) then
+         PosOffset = [0.0_ReKi, 0.0_ReKi, p%rotors(iWT)%WtrDpth]
+      else
+         PosOffset = 0.0_ReKi
+      end if
+
+      ! Hub
+      if (u%rotors(iWT)%HubMotion%Committed) then
+         call IfW_FlowField_GetVelAcc(p%FlowField, StartNode, t, &
+            real(u%rotors(iWT)%HubMotion%TranslationDisp + u%rotors(iWT)%HubMotion%Position, ReKi), &
```

Comment:
I like the idea of providing an interface for IfW_FlowField_GetVelAcc, it could solve some issues that came up during the integration which made it seem hard to use. I'll play around with it a little and do another commit.

Reply:
I took a look at providing multiple interfaces to IfW_GetVelAcc, but it requires declaring whether Position, Velocity, and Accel are always R4Ki or R8Ki. This would require many changes throughout the code. I'll keep thinking about a better solution as I continue to integrate the Flow Field.